.github/workflows/lint.yml

```diff
@@ -0,0 +1,33 @@
+name: Lint
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
```

Comment:
We should also add 
```
types: [opened, reopened, synchronize]
```

Recommend checking out https://github.com/apple/swift-collections/pull/426/files

Reply:
switched to match collections